test/e2e/scheduling/events.go

```diff
@@ -82,7 +83,9 @@ func observeEventAfterAction(c clientset.Interface, ns string, eventPredicate fu
 			AddFunc: func(obj interface{}) {
 				e, ok := obj.(*v1.Event)
 				ginkgo.By(fmt.Sprintf("Considering event: \nType = [%s], Name = [%s], Reason = [%s], Message = [%s]", e.Type, e.Name, e.Reason, e.Message))
-				framework.ExpectEqual(ok, true)
+				if !ok {
+					framework.Failf("event ", e.Name, " not found")
```

Comment:
`Failf` takes a format string, not a list of strings.

Also, `e` will be nil if `ok` is false. The `gingko.By` should crash. We need to move the check before it and *not* print `e`. Instead, use `framework.Failf("Expected *v1.Event, got %T %v", obj, obj)`.

Probably this never happens in practice, but we should still fix the test.

Reply:
Noted!